### ISSUE TITLE
Helm rollback to last successful release

### ISSUE THREAD

**Issue by tallpauley** [helm/helm#1796]
It would be nice to have `helm rollback LAST` or something of the like. Or it could be even fancier like git where you could do math `helm rollback LATEST~1` or something of the like.

**Comment by thomastaylor312** [comment]
It should probably follow something similar to `kubectl rollout`:
```
# Rollback to the previous deployment
kubectl rollout undo deployment/abc
  
# Rollback to deployment revision 3
kubectl rollout undo deployment/abc --to-revision=3
```

**Comment by technosophos** [comment]
I kinda like the Git symbolic notation.

**Comment by thomastaylor312** [comment]
I actually like it too. My thought for using something similar for `kubectl rollout` was to maintain consistency, but I don't think it is a huge deal 😄

**Comment by technosophos** [comment]
Totally. I think the `--to-revision` behavior is already present, though. Unless I'm misunderstanding what that does. I do like to stay consistent with kubectl when possible.

**Comment by technosophos** [comment]
I think that at least the core functionality of `helm rollback LAST` is fairly trivial, so I am marking this as a starter

Just to summarize where I think we are:
Today you can do `helm rollback MyRelease 2` to rollback to version 2. But it would be nice to have a shorthand way of just rolling back to the last version _that was not a FAILED release_.

**Comment by bcorijn** [comment]
@technosophos I had this vague recollection that the `helm rollback LAST` functionality was already implemented, but I could not find it back in the documentation. I dug a bit further and by reading the code, I learned it indeed is implented [here](https://github.com/kubernetes/helm/blob/master/pkg/tiller/release_rollback.go#L78-L80).  
So doing a `helm rollback MyRelease 0` will roll back to the last version. This creates another release number though, so it will only work for one single rollback.  

Seems like this has been an undocumented feature since October!

**Comment by jondlm** [comment]
I can confirm that rolling back to revision `0` works.

`kubectl rollout undo deployments <name> --to-revision = 0` works the same way.

**Comment by bacongobbler** [comment]
I think I've mentioned it a few times in Slack... Not sure how this issue skipped past my view. Yes, this has been a feature since #1248 was first introduced a year ago from today :)

I recall a long time ago (as seen in #1248), calling `helm rollback <release-name>` would imply a revision of "0", causing a rollback to the previous release. It appears the version is now a mandatory argument for `helm rollback`. We should see if we can make that optional again.

**Comment by bcorijn** [comment]
Strange enough last time I tried this, Helm gave as output it rolled back successfully but I saw no changes in my cluster. It has been on my todo list to look into, but I was expecting it might have just been removed from Helm as this feature was never documented.

**Comment by awwithro** [comment]
IMO it'd be really useful to give an arbitrary tag to a release number and do a rollback to that tag. Something like `helm rollback tag <REVISION> 1.0.0` and `helm rollback <RELEASE> 1.0.0` In my case I'm using a helm chart that includes istio route rules for canarying between two releases. Each change of the weights is a separate helm release. With an arbitrary tag, I can tag the release prior to changing weights and be able to rollback to it regardless of how many times I've changed weights and without having to track how many releases have been made.

**Comment by mitchellmaler** [comment]
What is the status of this? Right now in my CICD pipeline if the install fails I have to run a script to get the last successful revision number to rollback too. Which then the rollback is automated on deployment failure.

**Comment by bacongobbler** [comment]
Afaik nobody in the community is working on this effort

**Comment by edwardstudy** [comment]
Nowadays, I just backup and restore Helm release config data. When I restore releases data, I should run `helm rollback release_name lastest_version` to recreate pods.

**Comment by bacongobbler** [comment]
fixed in #5228
